commit 7aa9efa50924a6f8249f413cff367b39b348db08
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Tue Oct 25 05:01:10 2022 +0200

    patch: luxmark3/luxmark3-luxcore-fix-build-with-system-opencl-cmake.patch

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c668e1f3..708e0e056 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ if (NOT OPENIMAGEIO_FOUND)
 	MESSAGE(FATAL_ERROR "--> Could not locate required OpenImageIO files - Please check ${OPENIMAGEIO_SEARCH_PATH}")
 endif()
 
-if (NOT OPENCL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
+if (NOT OpenCL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
 	MESSAGE(ERROR "--> Could not locate required OpenCL files, disabling OpenCL support - Please check ${OPENCL_SEARCH_PATH}")
 	SET(LUXRAYS_DISABLE_OPENCL 1)
 endif()
@@ -134,9 +134,9 @@ add_subdirectory(src/luxcore)
 # 	add_subdirectory(samples/luxcorescenedemo)
 # 	add_subdirectory(samples/luxcoreimplserializationdemo)
 # endif()
-# if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
-# 	add_subdirectory(samples/smallluxgpu4)
-# endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+#if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+#	add_subdirectory(samples/smallluxgpu4)
+#endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
 
 
 ################################################################################
diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index d245bb4a5..fb5531876 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -111,8 +111,8 @@ endif ()
 set(OPENCL_ROOT                "${OPENCL_SEARCH_PATH}")
 find_package(OpenCL)
 
-if (OPENCL_FOUND)
-	include_directories(SYSTEM ${OPENCL_INCLUDE_DIR} ${OPENCL_C_INCLUDE_DIR})
+if (OpenCL_FOUND)
+	include_directories(SYSTEM ${OpenCL_INCLUDE_DIR})
 endif ()
 
 # Intel Embree
diff --git a/samples/smallluxgpu4/CMakeLists.txt b/samples/smallluxgpu4/CMakeLists.txt
index 2b6d659a2..599671e70 100644
--- a/samples/smallluxgpu4/CMakeLists.txt
+++ b/samples/smallluxgpu4/CMakeLists.txt
@@ -30,9 +30,9 @@ set(SMALLLUXGPU_SRCS
 add_executable(slg4 ${SMALLLUXGPU_SRCS})
 
 if(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${OpenCL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 else(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${OpenCL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 endif(APPLE)
 
 # This instructs FREEGLUT to emit a pragma for the static version
diff --git a/src/luxrays/CMakeLists.txt b/src/luxrays/CMakeLists.txt
index 33de137fd..c489a0808 100644
--- a/src/luxrays/CMakeLists.txt
+++ b/src/luxrays/CMakeLists.txt
@@ -238,7 +238,7 @@ TARGET_LINK_LIBRARIES(luxrays ${OPENIMAGEIO_LIBRARIES})
 
 IF (NOT LUXRAYS_DISABLE_OPENCL)
 	TARGET_LINK_LIBRARIES(luxrays ${OPENGL_gl_LIBRARY})
-	TARGET_LINK_LIBRARIES(luxrays ${OPENCL_LIBRARIES})
+	TARGET_LINK_LIBRARIES(luxrays ${OpenCL_LIBRARIES})
 ENDIF(NOT LUXRAYS_DISABLE_OPENCL)
 
 ###########################################################################
